contrib/inventory/netbox.py

```diff
@@ -0,0 +1,344 @@
+
+
+def get_value_by_path(source_dict, key_path, ignore_key_error=False):
+    """Get key value from nested dict by path.
+
+    Args:
+        source_dict: The dict that we look into.
+        key_path: The path of key in dot notion. e.g. "parent_dict.child_dict.key_name"
+        ignore_key_error: Ignore KeyError if the key not found in provided path.
+
+    Returns:
+        If key found in provided path, it will be returned.
+        If not, None will be returned.
+    """
+
+    try:
+        key_output = reduce_path(source_dict, key_path)
+    except KeyError as key_name:
+        if ignore_key_error:
+            key_output = None
+        else:
+            sys.exit("The key %s is not found. Please remember, Python is case sensitive." % key_name)
+    return key_output
+
+
+def open_yaml_file(yaml_file):
+    """Open YAML file.
+
+    Args:
+        yaml_file: Relative or absolute path to YAML file.
+
+    Returns:
+        Content of YAML the file.
+    """
+
+    # Load content of YAML file.
+    try:
+        with open(yaml_file, 'r') as config_yaml_file:
+            try:
+                yaml_file_content = yaml.safe_load(config_yaml_file)
+            except yaml.YAMLError as yaml_error:
+                sys.exit(yaml_error)
+    except IOError as io_error:
+        sys.exit("Cannot open YAML file.\n%s" % io_error)
+    return yaml_file_content
+
+
+class NetboxAsInventory(object):
+    """Netbox as a dynamic inventory for Ansible.
+
+    Retrieves hosts list from netbox API and returns Ansible dynamic inventory (JSON).
+
+    Attributes:
+        script_config_data: Content of its config which comes from YAML file.
+    """
+
+    def __init__(self, script_args, script_config_data):
+        # Script arguments.
+        self.config_file = script_args.config_file
+        self.list = script_args.list
+        self.host = script_args.host
+
+        # Script configuration.
+        main_config_key = "netbox"
+        script_config = script_config_data.get(main_config_key)
+        if script_config:
+            self.api_url = script_config["main"].get('api_url')
+            self.group_by = script_config.setdefault("group_by", {})
+            self.hosts_vars = script_config.setdefault("hosts_vars", {})
+        else:
+            sys.exit("The key '%s' is not found in config file." % main_config_key)
+
+        # Get value based on key.
+        self.key_map = {
+            "default": "name",
+            "general": "name",
+            "custom": "value",
+            "ip": "address"
+        }
+
+    @staticmethod
+    def get_hosts_list(api_url, specific_host=None):
+        """Retrieves hosts list from netbox API.
+
+        Returns:
+            A list of all hosts from netbox API.
+        """
+
+        if not api_url:
+            sys.exit("Please check API URL in script configuration file.")
+
+        if specific_host:
+            data_source = "{}?name={}".format(api_url, specific_host)
```

Comment:
I think it would be better to use ``params`` with ``requests.get`` rather than building url query.